babel/src/board.js

```diff
@@ -284,6 +284,18 @@ export class Board {
         return hasWideStyleMatch || _.some(_.map(chunks), hasMatchInPairOfRows);
     }
 
+    swap(swapOrbs) {
+        let [[x1, y1], [x2, y2]] = swapOrbs;
+        let orbsBefore = _.cloneDeep(this.orbs);
```

Comment:
Good lesson. It's called [reference semantics](http://legacy.python.org/doc/essays/ppt/acm-ws/sld039.htm) and is a great primer on how to avoid a lot of crazy bugs in many languages.

```py
def my_function(default_argument=[]): #  this is bad!!
    pass
```

`default_argument` will be mutated by all references made to it over time, even if it's through another variable that happens to be pointing to it.

```py
def my_function(default_argument=None):
    if default_argument is None:
        default_argument = []; #  this is the correct way to do it
```